crowdsourcehinter/crowdsourcehinter.py

```diff
@@ -0,0 +1,352 @@
+import ast
+import logging
+import operator
+import pkg_resources
+import random
+import json
+import copy
+
+from xblock.core import XBlock
+from xblock.fields import Scope, Dict, List, Boolean, String
+from xblock.fragment import Fragment
+
+log = logging.getLogger(__name__)
+
+
+class CrowdsourceHinter(XBlock):
+    """
+    This is the Crowdsource Hinter XBlock. This Xblock seeks to provide students with hints
+    that specifically address their mistake. Additionally, the hints that this Xblock shows
+    are created by the students themselves. This doc string will probably be edited later.
+    """
+    # Database of hints. hints are stored as such: {"incorrect_answer": {"hint": rating}}. each key (incorrect answer)
```

Comment:
Micro: this section of the code would be more readable with blank lines separating the fields from each other: one blank line before each block comment.